src/controllers/bulk/bulksupported.h

```diff
@@ -1,18 +1,34 @@
+#pragma once
+
+#include <cstdint>
+
 // A list of supported USB bulk devices
 
-#pragma once
+static constexpr std::uint32_t kInvalidInterfaceNumber = 0;
```

Comment:
Are you sure that 0 is an invalid interface number? AFAIK the USB interface is counted zero based.

Reply:
I didn't look too much into it either, but thats what I gathered from this code
https://github.com/mixxxdj/mixxx/blob/b2ff633773defdf6ede84c3a9349bba87ea4160b/src/controllers/bulk/bulkcontroller.cpp#L180-L203

But thats also defined on windows, even though `libusb_kernel_driver_active` is [not available on windows](https://libusb.sourceforge.io/api-1.0/group__libusb__dev.html#ga1cabd4660a274f715eeb82de112e0779)... so I'm not sure in how good of a shape that code is...

Reply:
The code for kernel driver unloading was introduced in #13008. According to the documentation linked by you, this is neither functional nor necessary on Windows.
But it's seems to be a validated argument in BulkController::open(). There 0 means the first interface of the USB device and not InvalidInterface number.
While Bulk is a part of the transport layer (OSI Level 4) of USB, the HID protocol is in the application layer (OSI level 7). This is why HID can use other transport layers than USB. And this is the reason why the USB interface number can be invalid for HID, but never for USB Bulk.

Reply:
right, but that doesn't explain why the interface logic is operating-system specific...

Reply:
Indeed, but my remark was that `kInvalidInterfaceNumber` does not match here.

Reply:
I agree. I removed the named constant.